src/behaviours/update-app/downgrading-version-update.test.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import type { ApplicationBuilder } from "../../renderer/components/test-utils/get-application-builder";
+import { getApplicationBuilder } from "../../renderer/components/test-utils/get-application-builder";
+import electronUpdaterIsActiveInjectable from "../../main/electron-app/features/electron-updater-is-active.injectable";
+import publishIsConfiguredInjectable from "../../main/update-app/publish-is-configured.injectable";
+import type { AsyncFnMock } from "@async-fn/jest";
+import asyncFn from "@async-fn/jest";
+import type { CheckForPlatformUpdates } from "../../main/update-app/check-for-platform-updates/check-for-platform-updates.injectable";
+import checkForPlatformUpdatesInjectable from "../../main/update-app/check-for-platform-updates/check-for-platform-updates.injectable";
+import checkForUpdatesInjectable from "../../main/update-app/check-for-updates/check-for-updates.injectable";
+import selectedUpdateChannelInjectable from "../../main/update-app/selected-update-channel.injectable";
+import type { DiContainer } from "@ogre-tools/injectable";
+import appVersionInjectable from "../../common/get-configuration-file-model/app-version/app-version.injectable";
+import { updateChannels } from "../../main/update-app/update-channels";
+
+describe("downgrading version update", () => {
+  let applicationBuilder: ApplicationBuilder;
+  let checkForPlatformUpdatesMock: AsyncFnMock<CheckForPlatformUpdates>;
+  let mainDi: DiContainer;
+
+  beforeEach(() => {
+    jest.useFakeTimers();
+
+    applicationBuilder = getApplicationBuilder();
+
+    applicationBuilder.beforeApplicationStart(({ mainDi }) => {
+      checkForPlatformUpdatesMock = asyncFn();
+
+      mainDi.override(
+        checkForPlatformUpdatesInjectable,
+        () => checkForPlatformUpdatesMock,
+      );
+
+      mainDi.override(electronUpdaterIsActiveInjectable, () => true);
+      mainDi.override(publishIsConfiguredInjectable, () => true);
+    });
+
+    mainDi = applicationBuilder.dis.mainDi;
+  });
+
+  [
+    {
+      updateChannel: updateChannels.latest,
+      appVersion: "4.0.0-beta",
+      downgradeIsAllowed: true,
```

Comment:
Shouldn't this depend on the version that is found?

Reply:
This tests the exact behaviour of the old implementation which was:

```typescript
@computed get isAllowedToDowngrade() {
    return new SemVer(getAppVersion()).prerelease[0] !== this.updateChannel;
  }
```

If that's wrong, sure. How would you describe it as a unit test?

Reply:
Fair enough.